components/bl-transferList-component/src/transfer-buttons.js

```diff
@@ -0,0 +1,148 @@
+export const TransferButtons = props => {
+  const { left, right, leftSelected, rightSelected, enableMoveAll } = props;
+
+  const { moveAllToRight, moveAllToLeft, moveSelectedToLeft, moveSelectedToRight } = useTransferActions(props);
+
+  return (
+    <div className="control-buttons">
+      { enableMoveAll &&
+        <button
+          onClick={ moveAllToRight }
+          disabled={ !left.length }
+          className="control-buttons__item">
+          { MoveIcons.DOUBLE_RIGHT }
+        </button>
+      }
+      <button
+        onClick={ moveSelectedToRight }
+        disabled={ !leftSelected.length }
+        className="control-buttons__item">
+        { MoveIcons.RIGHT }
+      </button>
+      <button
+        onClick={ moveSelectedToLeft }
+        disabled={ !rightSelected.length }
+        className="control-buttons__item">
+        { MoveIcons.LEFT }
+      </button>
+      { enableMoveAll &&
+        <button
+          onClick={ moveAllToLeft }
+          disabled={ !right.length }
+          className="control-buttons__item">
+          { MoveIcons.DOUBLE_LEFT }
+        </button>
+      }
+    </div>
+  );
+};
+
+const MoveIcons = {
+  LEFT: '<',
+  DOUBLE_LEFT: '≪',
+  RIGHT: '>',
+  DOUBLE_RIGHT: '≫',
+};
+
+const useTransferActions = props => {
```

Comment:
need to split the function, or I assume it can be split onto separated files, where each file contain its own button a logic for changing state